hazelcast/src/main/java/com/hazelcast/spi/impl/AbstractInvocationFuture.java

```diff
@@ -261,6 +262,9 @@ public void run() {
 
     // this method should not be needed; but there is a difference between client and server how it handles async throwables
     protected Throwable unwrap(Throwable throwable) {
+        if (throwable instanceof MemberLeftException) {
+            return throwable;
+        }
         if (throwable instanceof ExecutionException) {
             return throwable.getCause();
```

Comment:
wouldn't be better to have `if (throwable instanceof ExecutionException && throwable.getCause() != null))` instead?